SecretManagement.KeePass.Extension/Tests/Test-SecretVault.Tests.ps1

```diff
@@ -102,29 +142,509 @@ InModuleScope -ModuleName 'SecretManagement.KeePass.Extension' {
                 { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
             }
             It "should request a credential on the first pass" {
                 Test-SecretVault -VaultName $VaultName
-                Assert-MockCalled -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+                Should -Invoke -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+            }
+            It 'Should not request a credential on the second pass' {
+                Test-SecretVault -VaultName $VaultName
+                Should -Invoke -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+            }
+            It "should have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Not -Throw
+            }
+        }
+        Context "Validating with Path and incorrect UseMasterPassword" {
+            BeforeAll {
+                $MasterKey = 'You can not enter with this masterkey'
+                $VaultMasterKey = [PSCredential]::new('vaultkey', (ConvertTo-SecureString -AsPlainText -Force $MasterKey))
+
+                $VaultName = "KeepassPesterTest_$([guid]::NewGuid())"
+                $KeePassDatabaseSuffix = 'PathOnly'
+                $KeePassDatabaseFileName = "$($BaseKeepassDatabaseName)$($KeePassDatabaseSuffix).kdbx"
+                $VaultPath = Join-Path -Path $TestDrive -ChildPath $KeePassDatabaseFileName
+                Copy-Item -Path "$($PSScriptRoot)/$($KeePassDatabaseFileName)" -Destination $VaultPath
+
+                $RegisterSecretVaultPathOnlyParams = @{
+                    Name            = $VaultName
+                    ModuleName      = $ModulePath
+                    PassThru        = $true
+                    VaultParameters = @{
+                        Path = $VaultPath
+                    }
+                }
+                Microsoft.PowerShell.SecretManagement\Register-SecretVault @RegisterSecretVaultPathOnlyParams | Out-Null
+
+                Mock -Verifiable -CommandName 'Get-Credential' -MockWith { $VaultMasterKey }
+            }
+            AfterAll {
+                try {
+                    Microsoft.PowerShell.SecretManagement\Get-SecretVault -Name $VaultName -ErrorAction SilentlyContinue | Microsoft.PowerShell.SecretManagement\Unregister-SecretVault -ErrorAction SilentlyContinue
+                } catch [system.Exception] { }
+            }
+            It "should not have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
+            }
+            It "should throw on the first pass" {
+                { Test-SecretVault -VaultName $VaultName } | should -Throw
+            }
+            It "should still not have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
```

Comment:
-Throw should be relatively explicit to the expected exception, in this case: `| Should -Throw 'Cannot find a variable with the name*'`